pyramid_orb/services/model.py

```diff
@@ -97,17 +101,29 @@ def _update(self):
         record = self.model(self.record_id, context=context)
         record.update(values)
         record.save()
-        return record.__json__()
+        return record
+
+    def collect_actions_from_model(self, model):
+        actions = {}
+        for action, func in iter_actions(model):
+            actions[(action.name, action.method)] = func
+        return actions
 
     def get(self):
         values, context = get_context(self.request, model=self.model)
         if context.returning == 'schema':
             return self.model.schema()
         elif self.record_id:
             try:
-                return self.model(self.record_id, context=context)
+                record = self.model(self.record_id, context=context)
             except orb.errors.RecordNotFound:
                 raise HTTPNotFound()
+            action = self.get_action()
+            if record is not None and action is not None:
+                return action(record, self.request)
+            else:
+                return record
+
         else:
```

Comment:
we should also support classmethod actions, so that you can register an action to be performed on the model vs. on a record.  if `self.record_id` is not defined, then this endpoint acts as the action for the model class as a whole.  That would require checking to see if the action is an instance method or not